### PR TITLE
SEC-529: Add getTransactionObj to the archiver

### PR DESCRIPTION
**Pull request by S0naliThakur**
Linear: https://linear.app/shm/issue/SEC-526/
Summary: add `getTransactionObj`

### CODE REVIEW

**Review by abdulazeem-tk4vr — APPROVED** [review]

File: src/Data/ExtractShardKey.ts [hunk 1]
```diff
@@ -168,3 +170,26 @@ export async function extractKeyFromTx(receiptTx: any): Promise<string> {
   // return transformedSourceKey // executionShardKey
   return 'placeholder'
 }
+
+export function getTransactionObj(tx: any): Transaction[TransactionType.Legacy] | Transaction[TransactionType.AccessListEIP2930] {
+  if (!tx.raw) throw Error('fail')
```
Comment (aniketdivekar) [inline src/Data/ExtractShardKey.ts:175]: why so generic? is it copy pasted method?

File: src/Data/ExtractShardKey.ts [hunk 2]
```diff
@@ -168,3 +170,26 @@ export async function extractKeyFromTx(receiptTx: any): Promise<string> {
...
+        transactionObj =
+          TransactionFactory.fromSerializedData<TransactionType.AccessListEIP2930>(serializedInput)
+      } catch (e) {
+        if (config.VERBOSE) console.log('Unable to get transaction obj', e)
+      }
+    }
+  
+    if (transactionObj) {
+      return transactionObj
+    } else throw Error('tx obj fail')
```
Comment (aniketdivekar) [inline src/Data/ExtractShardKey.ts:194]: same here...

File: src/Data/ExtractShardKey.ts (hunk 1 above)
Comment (S0naliThakur) [reply src/Data/ExtractShardKey.ts:175]: yes

File: src/Data/ExtractShardKey.ts (hunk 2 above)
Comment (S0naliThakur) [reply src/Data/ExtractShardKey.ts:194]: this one also

**Pull request closed without merging**